### PR TITLE
Allow reservations that end in the future to be refunded

### PR DESCRIPTION
**Pull request by ranta**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Allow reservations that end in the future to be refunded

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3881](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3881)

[TILA-3881]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3881?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/tests/test_graphql_api/test_reservation/test_refund.py [hunk 1]
```diff
@@ -85,6 +85,11 @@ def test_reservation__refund__invalid_state__ends_in_the_future(graphql):
 @freeze_time("2022-01-01 12:00:00")
 @patch_method(VerkkokauppaAPIClient.refund_order, return_value=REFUND)
 def test_reservation__refund__correct_state__ends_in_the_future(graphql):
+    """
+    Reservations that end in the future can be refunded.
+    This was previously not allowed, but the restriction was removed in TILA-3881.
+    This test is kept as a regression test.
+    """
```
Comment (matti-lamppu) [inline backend/tests/test_graphql_api/test_reservation/test_refund.py:92]: I don't think messages like "this is how things worked in the past" are very useful. The current state of the code is how the system behaves, and this is just an edge case now — the test's name and contents reflect that already.

**Review by matti-lamppu — APPROVED** [review]

**Review by vergama — APPROVED** [review]
